marine/marine.py

```diff
@@ -186,11 +184,12 @@ def validate_fields(
         return bool(self._marine.validate_fields(fields_c_arr, fields_len))
 
     @staticmethod
-    def _parse_output(output: str) -> List[Optional[str]]:
+    def _parse_output(output: List[str], length: int) -> List[Optional[str]]:
         # TODO: this is a bottleneck. Find a better way to provide output from the c code
```

Comment:
Is this still a bottleneck? How much faster is this code now?

Reply:
On average the code is 5% faster.
In some cases (in tests without fields) we slowed the test a bit - might be some overhead we are paying for.